ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -80,6 +80,15 @@ public class Worker extends RemoteCompactorThread implements MetaStoreThread {
   static final private long SLEEP_TIME = 10000;
 
   private String workerName;
+  private final CompactorFactory compactorFactory;
+
+  public Worker() {
+    compactorFactory = CompactorFactory.getInstance();
+  }
+
+  public Worker(CompactorFactory compactorFactory) {
+    this.compactorFactory = compactorFactory;
```

Comment:
is this needed for testing only? if yes, compactorFactory could be replaced in test via the reflection

Reply:
It was done that way before, @kasakrisz requested to add a constructor instead. 

Personally I rather agree with him, on the long run I would like to eliminate usage of static methods and singletons  in compaction related classes whenever possible, and replace them with instance fields. Providing dependencies via constructor would make testing easier, and would allow devs to discover and control class dependencies much easier. It would also allow to introduce dependency injection easier in the future. I doubt it will happen but hope dies last :)